Frontend/src/pages/AddProposal.tsx

```diff
@@ -0,0 +1,115 @@
+import {
+  Button,
+  FormControl,
+  FormHelperText,
+  FormLabel,
+  Heading,
+  Input,
+  NumberDecrementStepper,
+  NumberIncrementStepper,
+  NumberInput,
+  NumberInputField,
+  NumberInputStepper,
+  Spinner,
+} from "@chakra-ui/react"
+import { useContext, useState } from "react"
+import { ContentWrapper } from "../components/PageWrapper"
+import StateContext from "../state/stateContext"
+
+interface Props {}
+
+export function AddProposal(_props: Props) {
+  const [input, setInput] = useState<any>({
+    collectionSize: "",
+    ipfsFolderCID: "",
+  })
+  const [isError, setIsError] = useState({
+    collectionSize: false,
+    ipfsFolderCID: false,
+  })
+
+  const [isSubmitting, setIsSubmitting] = useState<boolean>(false)
+
+  const { daoContract } = useContext(StateContext)
+
+  async function onSubmit() {
+    if (!daoContract) return
+
+    if (!input.collectionSize) {
+      setIsError({ ...isError, collectionSize: true })
+      return
+    }
+    if (!input.ipfsFolderCID) {
+      setIsError({ ...isError, ipfsFolderCID: true })
+      return
+    }
+    setIsError({ collectionSize: false, ipfsFolderCID: false })
+
+    setIsSubmitting(true)
+    await daoContract.addProposal(
+      input.collectionSize,
```

Comment:
What do we need `collectionSize` for, actually?

Reply:
In the Ballot
  
  struct Proposal {
    uint256 index; // index for the NFT collection
    uint256 voteCount; // number of accumulated votes
    uint256 collectionSize; // number of items in collection starting from 0
    bool active; // the proposal is still being voted on
    string ipfsFolderCID;
  }